libecs/include/pivot/ecs/Core/Component/SynchronizedComponentArray.hxx

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <mutex>
+
+#include "pivot/ecs/Core/Component/DenseComponentArray.hxx"
+#include "pivot/pivot.hxx"
+
+namespace pivot::ecs::component
+{
+
+/** \copybrief  DenseTypedComponentArray
+ *
+ * This child class of the DenseTypedComponentArray allow the array to be safely shared between thread.
+ */
+template <typename T, typename Mutex = std::mutex>
+class SynchronizedTypedComponentArray : public DenseTypedComponentArray<T>
```

Comment:
You don't want users of the class to access the base class without locking !
```suggestion
class SynchronizedTypedComponentArray : private DenseTypedComponentArray<T>
```

Reply:
Can't do with inheritance, because the IComponentArray inheritance need to be public. But good idea, I'll do something like that